### PR TITLE
EVG-2261 Rewrite merge test results without projection

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/task/db.go [hunk 1]
```diff
@@ -610,10 +561,9 @@ func FindOneNoMerge(query db.Q) (*Task, error) {
...
 func FindOne(query db.Q) (*Task, error) {
-	task := &Task{}
-	err := db.FindOneQ(Collection, query, task)
-	if err == mgo.ErrNotFound {
-		return nil, nil
+	task, err := FindOneNoMerge(query)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding task")
 	}
 	if err = task.MergeNewTestResults(); err != nil {
```
Comment (john-m-liu) [inline model/task/db.go]: should check for nil task here

File: model/task/db.go [hunk 2]
```diff
@@ -636,13 +586,22 @@ func FindOneId(id string) (*Task, error) {
...
+	return task, err
+}
+
+// FindOneOld returns one task from the old tasks collection that satisfies the query.
+func FindOneOld(query db.Q) (*Task, error) {
+	task, err := FindOneOldNoMerge(query)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding task")
+	}
 	if err = task.MergeNewTestResults(); err != nil {
```
Comment (john-m-liu) [inline model/task/db.go]: this should also check for a nil task

File: model/task/db.go (hunk 1 above)
Comment (bsamek) [reply model/task/db.go]: done

File: model/task/db.go (hunk 2 above)
Comment (bsamek) [reply model/task/db.go]: done

**Pull request merged**